python/tests/unit_tests/test_run_helpers.py

```diff
@@ -1201,14 +1201,14 @@ def my_run(foo: str):
 
     my_run(foo="bar", langsmith_extra={"parent": headers, "client": mock_client})
     mock_calls = _get_calls(mock_client)
-    assert len(mock_calls) == 1
```

Comment:
@hinthornw not sure i understand this behavior, do we need/want to maintain it?

Reply:
Ya. At one point what was happening was we'd create a RunTree from headers withoutt he overriding client, and then use that, meaning the user configuration was broken